### ISSUE TITLE
Connection-Probleme

### ISSUE THREAD

**Issue by OSch24** [microsoft/azuredatastudio#25377]
Type: <b>Performance Issue</b>

Hello everyone, first of all, I really enjoy working almost exclusively with ADS. I think the programme is great and would like to thank everyone who works with it.

I would just like to report a "problem" that occurs from time to time, but unfortunately I can't pinpoint why this happens.

This happens from time to time when I want to execute an SQL code or when I want to connect to an SQL server. Not always, but several times a day.

ADS really takes forever to connect to the server and execute a select top 10 * from table. This is not always ... I can sometimes work for hours and everything is great ... and then I want to start the sql again with F5 and it needs and needs and needs ... this sometimes takes up to 3 minutes ... then it suddenly works again.

If I switch to SQL Studio in the meantime, it works immediately. Sometimes I have SQL Studio open and ADS ... If it doesn't work with ADS again, I immediately test it on MSSMS and it always works there. I know it doesn't really help, but is there anything I can do?

mfg Oliver Schneider

Azure Data Studio version: azuredatastudio 1.47.1 (b6f7beb01f92adaa4b79b6b6f3ac704e95cafe6e, 2024-01-10T02:15:02.628Z)
OS version: Windows_NT x64 10.0.19045
Restricted Mode: No
Preview Features: Enabled
Modes:

**Comment by cheenamalhotra** [comment]
Hi @OSch24 

Please gather and upload logs as per instructions and mask any server info from logs as needed.

**Comment by OSch24** [comment]
> We need more info to debug your particular issue. If you could attach your logs to the issue (ensure no private data is in them), it would help us fix the issue much faster.
 

ok all clear.
I'm just running all the logs once.
But the console logs are also available in the Application Log folder.

I only have one problem, the log also contains our server names and path names, which I am not allowed to give out by the company.

Can I set somewhere that the path / server names are not logged?
Or only anonymised?

**Comment by cheenamalhotra** [comment]
I don't think this is possible in automated way, there's only one file "mssql.log" we need, where you can find and replace server connection info and share the file.

**Comment by OSch24** [comment]
ok thank you very much, then I will do this.

**Comment by OSch24** [comment]
Hi, I looked again, there is no "mssql.log" in my system ... but a folder with "Microsoft.mssql" and there are 3 files there ... 

The file "sqltools.log" looks interesting.


![image](https://github.com/microsoft/azuredatastudio/assets/48675132/5bb154ab-4617-43f6-a984-7680ebb967db)

**Comment by OSch24** [comment]
Now I have such a "hanger" again ... I have the command 

"SearchInAll 12" command.
This is a procedure that shows me a few results.

Timestamp:
24-02-25 16:58:09.9926389

Now after 13 minutes I cancel the process ... Normally this select takes about 0.1 sec.

If I use another server with sp_who2 to display everything, I don't see that a "different" select is hanging or anything like that ... everything is completely normal.

Enclosed is a log file.

[sqltools.log](https://github.com/microsoft/azuredatastudio/files/14396956/sqltools.log)

**Comment by cheenamalhotra** [comment]
Looking closer into logs, I see query is executing find on time, but the call to ListDatabases maybe getting stuck and unresponsive leading to stuck results rendering.. 

```
24-02-25 16:58:09.9926389 pid:5880 tid:152 sqltools Verbose: 0 : eventTID:21632 SqlCommand.Set_CommandText | API | Object Id 14069, String Value = 'SearchInAll 12', Client Connection Id 5814284e-d955-4c36-9a05-f221750fc73e
...
24-02-25 16:58:09.9927501 pid:5880 tid:152 sqltools Verbose: 0 : eventTID:21632 SqlCommand.RunExecuteReaderTds | Info | Object Id 14069, Activity Id 6c04fb35-e46d-4447-bc54-a62d61b77953:1, Client Connection Id 5814284e-d955-4c36-9a05-f221750fc73e, Command executed as SQLBATCH, Command Text 'SearchInAll 12' 
...
24-02-25 16:58:10.0720886 pid:5880 tid:152 sqltools Information: 0 : Batch:'Batch Id:'0', Elapsed:'00:00:00.0794818', HasError:'False'' on Query:'untitled:SQLQuery_1' completed
24-02-25 16:58:10.0723337 pid:5880 tid:152 sqltools Verbose: 0 : Sending message of id[], of type[Event] and method[query/batchComplete]
24-02-25 16:58:10.0724623 pid:5880 tid:152 sqltools Information: 0 : Query:'untitled:SQLQuery_1' completed
24-02-25 16:58:10.0727633 pid:5880 tid:152 sqltools Verbose: 0 : Sending message of id[], of type[Event] and method[query/complete]
// Query and results completed here

// List Databases call received but seem to be stuck for few mins.
24-02-25 16:58:24.9766341 pid:5880 tid:12 sqltools Verbose: 0 : Received message with id[4458], of type[Request] and method[connection/listdatabases]
```

**Comment by OSch24** [comment]
Many thanks.

It's not always ... but sometimes it hangs and hangs and hangs ... What always helps is to switch to Mangement Studio or terminate ADS and reconnect.

Is there anything else I can do to fix the "problem"?

**Comment by MishraPriyanka1** [comment]
Hi,

I also face the same problem with Azure data Studio it gets stuck and then I have to close the Application and restart again then works fine.
Any suggestions to fix the problem?
Regards

**Comment by OSch24** [comment]
Just a quick status update, with the current version 1.50 the problem unfortunately still exists, if this info is of interest.
